test/CMakeLists.txt

```diff
@@ -57,15 +57,30 @@ add_custom_target(run-onedpl-tests
     DEPENDS build-onedpl-tests
     COMMENT "Build and run all oneDPL tests")
 
-macro(onedpl_add_test test_source_file switch_off_checked_iterators)
+macro(onedpl_add_test test_source_file switch_off_checked_iterators is_std_test)
 
     set(_build_type_in_lower "unknown")
     if (CMAKE_BUILD_TYPE)
         string(TOLOWER "${CMAKE_BUILD_TYPE}" _build_type_in_lower)
     endif()
 
-    get_filename_component(_test_name ${test_source_file} NAME)
-    string(REPLACE "\.cpp" "" _test_name ${_test_name})
+    if (${is_std_test})
+        #process std tests: target name equal to parent folders names plus cpp file name without extension
+        file(RELATIVE_PATH _test_name "${CMAKE_CURRENT_SOURCE_DIR}" "${test_source_file}")
+        string(REPLACE "\.cpp" "" _test_name ${_test_name})
+        string(REPLACE "/" ";" _test_name_list ${_test_name})
+        
+        #get parent folder name and test name
+        list(GET _test_name_list -2 parent_folder_name)
+        list(GET _test_name_list -1 test_name)
+
+        set(result_test_name "std-${parent_folder_name}-${test_name}")
+        string(REPLACE "-" "-" _test_name ${result_test_name})
+    else()
+        #process oneDPL tests: target name equal to cpp file name without extension
+        get_filename_component(_test_name ${test_source_file} NAME)
+        string(REPLACE "\.cpp" "" _test_name ${_test_name})
+    endif()
```

Comment:
Does this code construct unique target names from files with the same names using their path? If, so Is it possible to name these files uniquely to simplify the logic?

Reply:
Yes, exactly, But I preferer to generate unique name to avoid test names like test1.pass.cpp, ..., testN.pass.cpp and so on.
Additionally this test are copied from another repository and better to save their names I believe.